Gems/LmbrCentral/Code/Source/Builders/LuaBuilder/LuaBuilderWorker.cpp

```diff
@@ -43,12 +43,29 @@ namespace LuaBuilder
                 {
                     AZStd::string watchFolder;
                     AZ::Data::AssetInfo assetInfo;
-                    AZ::IO::Path path(dependency.m_dependencyPath);
-                    bool isLuaDependency = !path.HasExtension() || path.Extension() == ".lua" || path.Extension() == ".luac";
-                    auto sourcePath = path.ReplaceExtension(".lua");
+                    // carbonated begin
+                    AZ::IO::Path path(dependency.m_dependencyPath.starts_with("/")
+                        ? dependency.m_dependencyPath.substr(1) : dependency.m_dependencyPath); // path shoul not start from a path delimiter
 
-                    if (assetSystem->GetSourceInfoBySourcePath(sourcePath.c_str(), assetInfo, watchFolder)
-                        && assetInfo.m_assetId.IsValid())
+                    if (!path.HasExtension() || path.Extension() == ".luac") // assume that a file witout extension is a lua file, ".luac" is generated from ".lua"
+                    {
+                        path.ReplaceExtension(".lua");
+                    }
+                    else if (path.Extension() == ".dynamicslice") // ".dynamicslice" is generated from ".slice"
+                    {
+                        path.ReplaceExtension(".slice");
+                    }
+                    else if (path.Extension() == ".spawnable") // ".spawnable" is generated from ".prefab"
+                    {
+                        path.ReplaceExtension(".prefab");
+                    }
+                    // todo: replace ".xxx.streamingimage" --> ".xxx" (where xxx is png, jpg, tif, ...), it requires to determine an original image path
```

Comment:
Create a map that you can iterate, instead of if/else/if/else

Reply:
Replaced to map